### ISSUE TITLE
use __slots__

### ISSUE THREAD

**Issue by d0c-s4vage** [d0c-s4vage/pfp#41]
This _should_ lead to some performance gains, both with speed and memory-usage

**Comment by d0c-s4vage** [comment]
https://stackoverflow.com/questions/472000/usage-of-slots/28059785#28059785

**Comment by zachriggle** [comment]
Is PFP currently performance-limited?

**Comment by d0c-s4vage** [comment]
Not really, but I suspect there could be some performance gains by implementing this.
